web/src/connectors/task-annotator-connector/use-document-data-lazy-loading/use-annotations-lazy-loading.ts

```diff
@@ -0,0 +1,96 @@
+import { useState } from 'react';
+import { AnnotationsResponse, useLatestAnnotations } from 'api/hooks/annotations';
+import {
+    getPageNumbersToKeepInCache,
+    mergeCachedArraysBasedOnCachedNumbers
+} from '../task-annotator-utils';
+import { PageInfo } from 'api/typings';
+import { TAnnotationsLazyLoadingParams } from './types';
+import { useCachedDataPagesRange } from './use-cached-data-pages-range';
+
+export const useAnnotationsLazyLoading = (
+    {
+        task,
+        job,
+        jobId,
+        fileMetaInfo,
+        revisionId,
+        nextLoadingPageNumbers,
+        pageNumbers,
+        availableRenderedPagesRange,
+        nextLoadingPagesRange,
+        setSelectedAnnotation
+    }: TAnnotationsLazyLoadingParams,
+    { enabled: hookEnabled }: { enabled: boolean }
+) => {
+    const [latestAnnotationsResultData, setLatestAnnotationsResultData] =
+        useState<AnnotationsResponse>();
+    const getJobId = () => (task ? task.job.id : jobId);
+    const getFileId = () => (task ? task.file.id : fileMetaInfo?.id);
+
+    const { cachedPageIndexesRange, setCachedRange } = useCachedDataPagesRange();
+
+    const latestAnnotationsQuery = useLatestAnnotations(
+        {
+            jobId: getJobId(),
+            fileId: getFileId(),
+            revisionId,
+            pageNumbers: nextLoadingPageNumbers,
+            userId:
+                job?.validation_type === 'extensive_coverage' && !revisionId ? task?.user_id : ''
+        },
+        {
+            enabled:
+                hookEnabled &&
+                Boolean(task || job || revisionId) &&
+                nextLoadingPageNumbers.length > 0,
+            onSuccess(newData) {
+                setLatestAnnotationsResultData((oldData) => {
+                    setCachedRange(availableRenderedPagesRange, nextLoadingPagesRange);
+
+                    if (!oldData) {
+                        return newData;
+                    }
+
+                    const pageNumbersToKeepInCache = getPageNumbersToKeepInCache(
+                        pageNumbers,
+                        availableRenderedPagesRange
+                    );
```

Comment:
The main idea of this PR is to load the data for PDF file in lazy mode but to not keep all loaded data in cache. In case if we keep all this data we may face a situation that for big documents we may have a lot of cached data which will crash browser's tab. In such situation, we'll only keep some amount of cached data. It means that in case if we loaded some data for some PDF pages then it does NOT mean this data will not be refetched anymore.